### DISCUSSION TITLE
Adding PicoCMS

### DISCUSSION THREAD

**Discussion by DrZoidberg09** [nextcloud/all-in-one#1147] [Ideas] [answered]
Hi there,

I am actually not sure were to put it. However, the ultimate goal is to use PicoCMS with Nextcloud AIO.

To do this, the web server config need to edited to rewrite to the PicoCMS URL. In this case it is Apache/Caddy I think?

Would it be possible to add the rewrite options directly in the Caddyfile? And / or is it possible to manually edit this file?

Thank you so much!

**Comment by szaimen** [comment, marked as answer]
Hello, this is possible if you run AIO behind a reverse proxy. See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md.
You can then do the needed changes in the reverse proxy config.

**Reply by DrZoidberg09** [reply]
Yes, I know that this would work. However, we set-up the Nextcloud instance without reverse proxy (as it was meant to be in the early versions of Nextcloud AIO).

It would be only a small change e in the Caddyfile:

>   route {
        rewrite /sites /index.php/apps/cms_pico/pico_proxy
        reverse_proxy localhost:8000 
    }

I hope I translated the Apache line properly to Caddy... :)

The original code for Apache is:

> ProxyPass /sites/ {Nextcloud_URL}/index.php/apps/cms_pico/pico_proxy/
ProxyPassReverse /sites/ {Nextcloud_URL}/index.php/apps/cms_pico/pico_proxy/
SSLProxyEngine on

Would that be possible to add it to the Caddyfile?

That would be really nice!

**Reply by szaimen** [reply]
No, we will not add this. you can recreate the mastercontainer anytime and configure it to work behind a reverse proxy.
